spec/semantic_range_ltr_spec.rb

```diff
@@ -74,92 +74,98 @@
-    expect(SemanticRange.ltr?('2.1.3', '2.*.*')).to eq(false)
-    expect(SemanticRange.ltr?('1.2.3', '1.2.*')).to eq(false)
-    expect(SemanticRange.ltr?('2.1.3', '1.2.* || 2.*')).to eq(false)
-    expect(SemanticRange.ltr?('1.2.3', '1.2.* || 2.*')).to eq(false)
-    expect(SemanticRange.ltr?('1.2.3', '1.2.* || 2.*')).to eq(false)
-    expect(SemanticRange.ltr?('1.2.3', '*')).to eq(false)
-    expect(SemanticRange.ltr?('2.1.2', '2')).to eq(false)
-    expect(SemanticRange.ltr?('2.3.1', '2.3')).to eq(false)
-    expect(SemanticRange.ltr?('2.4.0', '~2.4')).to eq(false) # >=2.4.0 <2.5.))
-    expect(SemanticRange.ltr?('2.4.5', '~2.4')).to eq(false)
-    expect(SemanticRange.ltr?('3.2.2', '~>3.2.1')).to eq(false) # >=3.2.1 <3.3.))
-    expect(SemanticRange.ltr?('1.2.3', '~1')).to eq(false) # >=1.0.0 <2.0.))
-    expect(SemanticRange.ltr?('1.2.3', '~>1')).to eq(false)
-    expect(SemanticRange.ltr?('1.2.3', '~> 1')).to eq(false)
-    expect(SemanticRange.ltr?('1.0.2', '~1.0')).to eq(false) # >=1.0.0 <1.1.))
-    expect(SemanticRange.ltr?('1.0.2', '~ 1.0')).to eq(false)
-    expect(SemanticRange.ltr?('1.0.0', '>=1')).to eq(false)
-    expect(SemanticRange.ltr?('1.0.0', '>= 1')).to eq(false)
-    expect(SemanticRange.ltr?('1.1.1', '<1.2')).to eq(false)
-    expect(SemanticRange.ltr?('1.1.1', '< 1.2')).to eq(false)
-    expect(SemanticRange.ltr?('0.5.5', '~v0.5.4-pre')).to eq(false)
-    expect(SemanticRange.ltr?('0.5.4', '~v0.5.4-pre')).to eq(false)
-    expect(SemanticRange.ltr?('0.7.2', '=0.7.x')).to eq(false)
-    expect(SemanticRange.ltr?('0.7.2', '>=0.7.x')).to eq(false)
-    expect(SemanticRange.ltr?('0.6.2', '<=0.7.x')).to eq(false)
-    expect(SemanticRange.ltr?('0.2.5', '>0.2.3 >0.2.4 <=0.2.5')).to eq(false)
-    expect(SemanticRange.ltr?('0.2.4', '>=0.2.3 <=0.2.4')).to eq(false)
-    expect(SemanticRange.ltr?('2.0.0', '1.0.0 - 2.0.0')).to eq(false)
-    expect(SemanticRange.ltr?('4.0.0', '^3.0.0')).to eq(false)
-    expect(SemanticRange.ltr?('2.0.0', '^1.0.0 || ~2.0.1')).to eq(false)
-    expect(SemanticRange.ltr?('3.2.0', '^0.1.0 || ~3.0.1 || 5.0.0')).to eq(false)
-    expect(SemanticRange.ltr?('1.0.0beta', '^0.1.0 || ~3.0.1 || 5.0.0', loose: true)).to eq(false)
-    expect(SemanticRange.ltr?('5.0.0-0', '^0.1.0 || ~3.0.1 || 5.0.0', loose: true)).to eq(false)
-    expect(SemanticRange.ltr?('3.5.0', '^0.1.0 || ~3.0.1 || >4 <=5.0.0')).to eq(false)
-    expect(SemanticRange.ltr?('1.0.0beta', '^1.0.0alpha', loose: true)).to eq(false)
-    expect(SemanticRange.ltr?('1.0.0beta', '~1.0.0alpha', loose: true)).to eq(false)
-    expect(SemanticRange.ltr?('1.0.0beta', '^1.0.0-alpha', loose: true)).to eq(false)
-    expect(SemanticRange.ltr?('1.0.0beta', '~1.0.0-alpha', loose: true)).to eq(false)
-    expect(SemanticRange.ltr?('1.0.0-beta', '^1.0.0-alpha')).to eq(false)
-    expect(SemanticRange.ltr?('1.0.0-beta', '~1.0.0-alpha')).to eq(false)
-    expect(SemanticRange.ltr?('1.0.0', '=0.1.0')).to eq(false)
+    [['~ 1.0', '1.1.0'],
```

Comment:
This is pretty tough to grok the changes on, maybe do the cleanup and changes as separate commits?